File: sample/src/main/java/vn/tiki/sample/extra/ExtraInjectionFragment.java
package vn.tiki.sample.extra;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import intents.Intents;
import vn.tiki.intents.BindExtra;
import vn.tiki.sample.R;

public class ExtraInjectionFragment extends Fragment {

  @BindView(R.id.tvHeight) TextView tvHeight;
  @BindView(R.id.tvWeight) TextView tvWeight;

  @BindExtra int height;
  @BindExtra int weight;

  @Nullable @Override
  public View onCreateView(
      LayoutInflater inflater,
      @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    final View view = inflater.inflate(R.layout.fragment_extra_injection, container, false);

    ButterKnife.bind(this, view);
    Intents.bind(this);

    tvHeight.setText(height + " cm");
    tvWeight.setText(weight + " kg");

    return view;
  }
}

File: sample/src/main/java/vn/tiki/sample/extra/ExtraInjectionFragment.kt
package vn.tiki.sample.extra

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import kotlinx.android.synthetic.main.fragment_extra_injection.tvHeight
import kotlinx.android.synthetic.main.fragment_extra_injection.tvWeight
import vn.tiki.sample.Extras
import vn.tiki.sample.R
import vn.tiki.sample.util.extraOrElse

class ExtraInjectionFragment : Fragment() {
  private val height: Int by extraOrElse(Extras.HEIGHT, 0)
  private val weight: Int by extraOrElse(Extras.WEIGHT, 0)

  override fun onCreateView(
      inflater: LayoutInflater?,
      container: ViewGroup?,
      savedInstanceState: Bundle?): View? {
    return inflater?.inflate(R.layout.fragment_extra_injection, container, false)
  }

  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)

    ButterKnife.bind(this, view!!)

    tvHeight.text = "$height cm"
    tvWeight.text = "$weight kg"
  }
}
